e2etest/l2tests/interface_selector.go

```diff
@@ -107,36 +107,27 @@ var _ = ginkgo.Describe("L2-interface selector", func() {
 				Expect(err).NotTo(HaveOccurred())
 			}()
 
-			allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
-			Expect(err).NotTo(HaveOccurred())
-
-			var l2Status *metallbv1beta1.ServiceL2Status
 			Eventually(func() error {
-				var ss []*metallbv1beta1.ServiceL2Status
-				if ss, err = status.GetSvcPossibleL2Status(ConfigUpdater.Client(), svc, allNodes); err == nil {
-					if len(ss) != 1 {
-						return fmt.Errorf("")
-					}
-					l2Status = ss[0]
+				var statuses []metallbv1beta1.ServiceL2Status
+				if statuses, err = status.L2ForService(ConfigUpdater.Client(), svc); err != nil {
+					return err
 				}
-				return err
-			}, 2*time.Minute, 1*time.Second).ShouldNot(HaveOccurred())
-
-			Consistently(func() error {
-				l2Status, err = status.GetL2Status(ConfigUpdater.Client(), svc, l2Status.Status.Node)
-				if err != nil || len(l2Status.Status.Interfaces) == 0 {
-					return fmt.Errorf("")
+				if len(statuses) != 1 {
+					return fmt.Errorf("expected 1 ServiceL2Status object, got %d", len(statuses))
 				}
 				return nil
-			}, 5*time.Second).ShouldNot(HaveOccurred())
+			}, 2*time.Minute, 1*time.Second).ShouldNot(HaveOccurred())
 
-			Consistently(func() error {
-				l2Status, err = status.GetL2Status(ConfigUpdater.Client(), svc, l2Status.Status.Node)
-				if err != nil || l2Status.Status.Interfaces[0].Name != NodeNics[0] {
-					return fmt.Errorf("")
+			Consistently(func() string {
+				var ss []metallbv1beta1.ServiceL2Status
+				if ss, err = status.L2ForService(ConfigUpdater.Client(), svc); err != nil || len(ss) != 1 {
```

Comment:
let's be more descriptive when this fails, like we do above (I know it was already sub-optimal :-) )